zetaclient/chains/sui/signer/signer.go

```diff
@@ -201,6 +213,23 @@ func (s *Signer) SignTxWithCancel(
 	return sig, sigCancel, nil
 }
 
+// validSuiAddress checks whether the input string is a valid Sui address
+// For WithdrawAndCall, the receiver is the target package ID. It follows same format, so we use same validation for both
+func validSuiAddress(addr string) error {
```

Comment:
Can we use and modify https://github.com/zeta-chain/node/blob/05ba3daf0599ebfaa60c1909ac97000963943be8/pkg/contracts/sui/address.go#L44 instead

Reply:
This is done by intention. It is to prevent someone build the `zetacore` binary themself and run their validator on a forked logic.